### ISSUE TITLE
Multi-platform support (Kotlin/Native)

### ISSUE THREAD

**Issue by bwalter** [square/wire#880]
It would be awesome if Wire would support generation of Kotlin classes used in multi-platform project and have a runtime which supports both Android and iOS.

Based on the activity and the changes in okio, it seems that this goal is realistic. Do you have any concrete goal on that direction?

Thanks for the great work!

**Comment by swankjesse** [comment]
Yep, we're working on it.

**Comment by SAGARSURI** [comment]
I don’t wanna use Ktor as the network client. I am waiting for Wire to be the default network client for both multi-platform and native android app development.

**Comment by Egorand** [comment]
@SAGARSURI Wire is not a network client, it's a code generation library. Wire's gRPC implementation depends on OkHttp for networking on JVM and we currently don't have a Native networking backend.

**Comment by SAGARSURI** [comment]
Thanks for the clarification @Egorand

**Comment by bourquep** [comment]
Does Wire 3 support Kotlin multiplatform?

**Comment by Egorand** [comment]
@bourquep it does, check out the demo here: https://github.com/Egorand/wire-multiplatform-sample

**Comment by bourquep** [comment]
Thanks for this @Egorand! More specifically, does Wire 3 support gRpc (client) on multiple platforms?

**Comment by Egorand** [comment]
No, gRPC is currently only supported on JVM.

**Comment by bourquep** [comment]
Do you know if there are short-term plans to support it on other platforms?

**Comment by Egorand** [comment]
PRs are welcome!

**Comment by oldergod** [comment]
It is taking time but it's on our roadmap. We'll update issues when we have news.

**Comment by kaushalyap** [comment]
@swankjesse, @oldergod any progress on this?

**Comment by genemcahill** [comment]
How do you generate multi-platform Kotlin code using the compiler CLI?

**Comment by AlexITC** [comment]
Is there any update on supporting iOS?

**Comment by oldergod** [comment]
We don't have plans on adding gRPC support for iOS at the moment. Maybe one thing that Wire could do is to generate interfaces for iOS (or other) consumers to plug their http client into.

**Comment by deepakrathi** [comment]
does it now support gRpc on Kotlin multi-platforms? like shared for both Android/iOS platforms?

**Comment by oldergod** [comment]
No, it does not. `GrpcClient` is now an abstract class which can be implemented on all platforms (except the JVM-I'm working on it) if you'd like.

**Comment by sebaslogen** [comment]
To use wire in Kotlin mutliplatform you need native Android and iOS clients, since the wire generated models are already kotlin-only you can use them in the kotlin multiplatform module.
Here is a longer explanation of how to connect the wire generated code with the clients and a sample implementation:
https://github.com/grpc/grpc-kotlin/issues/51#issuecomment-1851574950
